crates/astria-core/src/sequencerblock/v1alpha1/celestia.rs

```diff
@@ -494,6 +494,17 @@ pub struct SubmittedMetadata {
     rollup_ids_proof: merkle::Proof,
 }
 
+/// An iterator over rollup IDs.
+pub struct RollupIdIter<'a>(std::slice::Iter<'a, RollupId>);
+
+impl<'a> Iterator for RollupIdIter<'a> {
+    type Item = &'a RollupId;
```

Comment:
`RollupId` are `Copy`, so this could be a `type Item = RollupId;`

Reply:
Yeah - I've had a similar debate a bunch of times in the past :D  Just because a type _is_ `Copy` I don't know if that means we should favour copying over passing by ref.  It make sense for very small types, but less so for larger types.  The question then boils down to: is a 32-byte array small enough?

I guess I've landed on generally going for pass-by-ref since it's slightly more performant, but it's a weak preference outside of hot paths.